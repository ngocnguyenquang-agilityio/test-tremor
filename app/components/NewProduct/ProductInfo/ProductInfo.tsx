"use client";

import dynamic from "next/dynamic";

// Libs
import { useForm, Controller } from "react-hook-form";

// Components
import { TextInput, Text, Flex, Select, SelectItem, Card } from "@tremor/react";
const QuillEditor = dynamic(() => import("react-quill"), { ssr: false });

// Constants
import { MESSAGES_ERROR, CATEGORY_PRODUCT, COLOR_PRODUCT } from "@/constants";

// Types
import { TProductInfo, SelectOptionData } from "@/types";

// Styles
import "@/styles/form.css";
import "react-quill/dist/quill.snow.css"; // Import Quill styles

interface ProductInfoData {
  productName?: string;
  description?: string;
  weight?: string;
  category?: string;
  quantity?: string;
  isEdit?: boolean;
  collection?: string;
  price?: string;
  color?: string;
}

const ProductInfo = ({
  productName,
  description,
  weight,
  category,
  quantity,
  isEdit = false,
  collection,
  price,
  color,
}: ProductInfoData) => {
  const {
    control,
    formState: { errors },
    handleSubmit,
  } = useForm<TProductInfo>({
    defaultValues: {
      name: productName,
      description: description,
      weight: weight,
      category: category,
      quantity: quantity,
      collection: collection,
      price: price,
      color: color,
    },
    mode: "onSubmit",
  });

  const { name } = errors || {};
  const nameErrorMessage = name?.message?.toString() || "";

  const handleNext = () => {
    //TODO handle to check submit form with next button
  };

  return (
    <Card className="w-full dark:bg-dark-tremor-primary rounded-lg shadow-box-icon-default ring-0">
      <Text className="text-primary dark:text-white font-bold text-xl mb-8">
        Product Information
      </Text>
      <form onSubmit={handleSubmit(handleNext)}>
        <Flex className="items-start gap-4">
          <Flex flexDirection="col" className="gap-4">
            <Controller
              control={control}
              rules={{
                required: MESSAGES_ERROR.NAME_REQUIRED,
                minLength: {
                  value: 4,
                  message: MESSAGES_ERROR.NAME_MIN_LENGTH,
                },
              }}
              render={({ field }) => (
                <div className="w-full mb-2">
                  <Text className="text-secondary dark:text-lighter mb-2">
                    Name
                  </Text>
                  <TextInput
                    id="name"
                    placeholder="Name"
                    className="py-1 w-full dark:text-white hover:bg-transparent bg-transparent dark:bg-transparent focus:bg-transparent rounded-b-none border-l-0 border-r-0 border-t-0 border-b-1 focus:border-b-2 focus:outline-none focus:border-tremor-brand-subtle dark:border-light dark:focus:border-white shadow-none hover:bg-transparent ring-0"
                    autoFocus
                    required
                    {...field}
                  />
                  {nameErrorMessage && (
                    <p className="pt-1 text-[11px] xs:text-xs text-red-500">
                      {nameErrorMessage}
                    </p>
                  )}
                </div>
              )}
              name="name"
            />
            {isEdit && (
              <Flex className="mb-4 gap-4">
                <Controller
                  control={control}
                  render={({ field }) => (
                    <div className="w-full">
                      <Text className="text-secondary dark:text-lighter mb-2">
                        Collection
                      </Text>
                      <TextInput
                        id="collection"
                        placeholder="Summer"
                        className="py-1 w-full dark:text-white hover:bg-transparent bg-transparent dark:bg-transparent focus:bg-transparent rounded-b-none border-l-0 border-r-0 border-t-0 border-b-1 focus:border-b-2 focus:outline-none focus:border-tremor-brand-subtle dark:border-light dark:focus:border-white shadow-none hover:bg-transparent ring-0"
                        {...field}
                      />
                    </div>
                  )}
                  name="collection"
                />
                <Controller
                  control={control}
                  render={({ field }) => (
                    <div className="w-full">
                      <Text className="text-secondary dark:text-lighter mb-2">
                        Price
                      </Text>
                      <TextInput
                        id="price"
                        placeholder="$99"
                        className="py-1 w-full dark:text-white hover:bg-transparent bg-transparent dark:bg-transparent focus:bg-transparent rounded-b-none border-l-0 border-r-0 border-t-0 border-b-1 focus:border-b-2 focus:outline-none focus:border-tremor-brand-subtle dark:border-light dark:focus:border-white shadow-none hover:bg-transparent ring-0"
                        {...field}
                      />
                    </div>
                  )}
                  name="price"
                />
              </Flex>
            )}
            <Controller
              control={control}
              render={({ field }) => (
                <div className="w-full">
                  <Text className="text-secondary dark:text-lighter mb-2">
                    Description &#x276A;optional&#x276B;
                  </Text>
                  <QuillEditor
                    placeholder="Content goes here..."
                    className="w-full rounded-md text-secondary dark:text-white"
                    {...field}
                  />
                </div>
              )}
              name="description"
            />
          </Flex>
          <Flex flexDirection="col">
            <Controller
              control={control}
              render={({ field }) => (
                <div className="w-full mb-6">
                  <Text className="text-secondary dark:text-lighter mb-2">
                    Weight
                  </Text>
                  <TextInput
                    id="weight"
                    placeholder="Weight"
                    className="py-1 w-full dark:text-white hover:bg-transparent bg-transparent dark:bg-transparent focus:bg-transparent rounded-b-none border-l-0 border-r-0 border-t-0 border-b-1 focus:border-b-2 focus:outline-none focus:border-tremor-brand-subtle dark:border-light dark:focus:border-white shadow-none hover:bg-transparent ring-0"
                    {...field}
                  />
                </div>
              )}
              name="weight"
            />

            <Controller
              control={control}
              render={({ field }) => (
                <div className="w-full mb-8">
                  <Text className="text-secondary dark:text-lighter mb-2">
                    Quantity
                  </Text>
                  <TextInput
                    id="quantity"
                    placeholder="Quantity"
                    className="py-1 w-full dark:text-white hover:bg-transparent bg-transparent dark:bg-transparent focus:bg-transparent rounded-b-none border-l-0 border-r-0 border-t-0 border-b-1 focus:border-b-2 focus:outline-none focus:border-tremor-brand-subtle dark:border-light dark:focus:border-white shadow-none hover:bg-transparent ring-0"
                    {...field}
                  />
                </div>
              )}
              name="quantity"
            />

            <Controller
              control={control}
              render={() => (
                <div className="w-full mb-7">
                  <Text className="text-secondary mb-3 dark:text-lighter mb-2">
                    Category
                  </Text>
                  <Select
                    placeholder="Clothing"
                    className="select-custom dark:text-white dark:border-light dark:focus:border-white">
                    {CATEGORY_PRODUCT.map((item: SelectOptionData) => (
                      <SelectItem key={item.value} value={item.value}>
                        {item.option}
                      </SelectItem>
                    ))}
                  </Select>
                </div>
              )}
              name="category"
            />
            {isEdit && (
              <Controller
                control={control}
                render={() => (
                  <div className="w-full mb-4">
                    <Text className="text-secondary mb-3 dark:text-lighter mb-2">
                      Color
                    </Text>
                    <Select
                      placeholder="Black"
                      className="select-custom dark:text-white dark:border-light dark:focus:border-white">
                      {COLOR_PRODUCT.map((item: SelectOptionData) => (
                        <SelectItem key={item.value} value={item.value}>
                          {item.option}
                        </SelectItem>
                      ))}
                    </Select>
                  </div>
                )}
                name="color"
              />
            )}
          </Flex>
        </Flex>
      </form>
    </Card>
  );
};

export default ProductInfo;
