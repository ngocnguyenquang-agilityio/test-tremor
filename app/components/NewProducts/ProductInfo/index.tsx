"use client";

// Libs
import { useForm, Controller } from "react-hook-form";
import { useState } from "react";
import dynamic from "next/dynamic";

// Components
import {
  TextInput,
  Button,
  Text,
  Flex,
  Select,
  SelectItem,
} from "@tremor/react";

// Constants
import { MESSAGES_ERROR } from "@/constants";

// Types
import { TProductInfo } from "@/types";

// Styles
import "@/styles/form.css";
import "react-quill/dist/quill.snow.css"; // Import Quill styles

const QuillEditor = dynamic(() => import("react-quill"), { ssr: false });

const ProductInfo = () => {
  const {
    control,
    formState: { errors },
    handleSubmit,
  } = useForm<TProductInfo>({
    defaultValues: {
      name: "",
      description: "",
      weight: "",
      category: "",
      size: "",
    },
    mode: "onSubmit",
  });

  const [formStatus, setFormStatus] = useState({
    isPending: false,
    errorMessage: "",
  });

  const { name } = errors || {};
  const nameErrorMessage = name?.message?.toString() || "";
  const isDisableSubmit = formStatus.isPending;

  const handleNext = () => {
    setFormStatus({
      isPending: false,
      errorMessage: nameErrorMessage,
    });
  };

  return (
    <form
      onSubmit={handleSubmit(handleNext)}
      className="w-full p-2 sm:p-4 bg-white dark:bg-dark-tremor-primary rounded-lg w-[67%] shadow-box-icon-default">
      <Text className="text-primary dark:text-white font-bold text-xl mb-8">
        Product Infomation
      </Text>
      <Flex className="flex-col items-end">
        <Flex className="items-start">
          <Flex className="flex-col">
            <Controller
              control={control}
              rules={{
                required: MESSAGES_ERROR.NAME_REQUIRED,
                minLength: {
                  value: 4,
                  message: MESSAGES_ERROR.NAME_MIN_LENGTH,
                },
              }}
              render={({ field }) => (
                <div className="h-[70px] w-full min-w-[290px]">
                  <TextInput
                    id="name"
                    placeholder="Name"
                    className="py-1 w-full dark:text-white hover:bg-transparent bg-transparent dark:bg-transparent focus:bg-transparent rounded-b-none border-l-0 border-r-0 border-t-0 border-b-1 focus:border-b-2 focus:outline-none focus:border-tremor-brand-subtle dark:border-light dark:focus:border-white shadow-none hover:bg-transparent ring-0"
                    autoFocus
                    required
                    {...field}
                  />
                  {nameErrorMessage && (
                    <p className="pt-1 text-[11px] xs:text-xs text-red-500">
                      {nameErrorMessage}
                    </p>
                  )}
                </div>
              )}
              name="name"
            />
            <Flex className="flex-col">
              <Flex className="justify-start mb-3">
                <Text className="text-secondary dark:text-white">
                  Description
                </Text>
                <Text className="text-xs ml-1 dark:text-white">(Optional)</Text>
              </Flex>
              <Controller
                control={control}
                render={({ field }) => (
                  <div className="w-full min-w-[290px]">
                    <QuillEditor
                      placeholder="Content goes here..."
                      className="w-full rounded-md dark:text-white"
                      {...field}
                    />
                  </div>
                )}
                name="description"
              />
            </Flex>
          </Flex>
          <Flex className="ml-6 flex-col">
            <Controller
              control={control}
              render={({ field }) => (
                <div className="h-[70px] w-full min-w-[290px]">
                  <TextInput
                    id="weight"
                    placeholder="Weight"
                    className="py-1 w-full dark:text-white hover:bg-transparent bg-transparent dark:bg-transparent focus:bg-transparent rounded-b-none border-l-0 border-r-0 border-t-0 border-b-1 focus:border-b-2 focus:outline-none focus:border-tremor-brand-subtle dark:border-light dark:focus:border-white shadow-none hover:bg-transparent ring-0"
                    {...field}
                  />
                </div>
              )}
              name="weight"
            />
            <Flex className="flex-col items-start">
              <Text className="text-secondary mb-3 dark:text-white">
                Category
              </Text>
              <Controller
                control={control}
                render={() => (
                  <div className="h-[70px] w-full">
                    <Select
                      placeholder="Clothing"
                      className="select-custtom dark:text-white dark:border-light dark:focus:border-white">
                      <SelectItem value="1">Clothing</SelectItem>
                      <SelectItem value="2">Electronics</SelectItem>
                      <SelectItem value="3">Furniture</SelectItem>
                      <SelectItem value="4">Others</SelectItem>
                      <SelectItem value="5">Real Estate</SelectItem>
                    </Select>
                  </div>
                )}
                name="category"
              />
            </Flex>
            <Flex className="flex-col items-start">
              <Text className="text-secondary mb-3 dark:text-white">Size</Text>
              <Controller
                control={control}
                render={() => (
                  <div className="h-[70px] w-full min-w-[290px]">
                    <Select
                      placeholder="Small"
                      className="select-custtom dark:text-white dark:bg-transparent">
                      <SelectItem value="1">Extra Large</SelectItem>
                      <SelectItem value="2">Extra Small</SelectItem>
                      <SelectItem value="3">Large</SelectItem>
                      <SelectItem value="4">Medium</SelectItem>
                      <SelectItem value="5">Small</SelectItem>
                    </Select>
                  </div>
                )}
                name="size"
              />
            </Flex>
          </Flex>
        </Flex>
        <Button
          aria-disabled={isDisableSubmit}
          type="submit"
          className="antialiased min-w-[80px] py-[12px] text-center uppercase sm:px-[21px] bg-gradient-primary dark:bg-gradient-pickled !shadow-btn-primary px-6 py-2.5 rounded-lg border-0 hover:!shadow-btn-primary-hover"
          size="xs"
          disabled={isDisableSubmit}>
          <Text className="items-center uppercase py-[2px] text-xs font-bold font-primary text-white dark:text-dark-tremor-content-title">
            Next
          </Text>
        </Button>
      </Flex>
    </form>
  );
};

export default ProductInfo;
