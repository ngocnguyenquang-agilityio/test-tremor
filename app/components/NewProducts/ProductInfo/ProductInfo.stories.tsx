import type { Meta, StoryObj } from "@storybook/react";

// Components
import { Flex } from "@tremor/react";
import ProductInfo from "./ProductInfo";

const meta = {
  title: "Components/ProductInfo",
  component: ProductInfo,
  tags: ["autodocs"],
} as Meta<typeof ProductInfo>;

export default meta;

type Story = StoryObj<typeof meta>;

export const ProductInfoDefault: Story = {
  render: () => (
    <Flex className="w-full justify-center mt-10">
      <ProductInfo />
    </Flex>
  ),
};
