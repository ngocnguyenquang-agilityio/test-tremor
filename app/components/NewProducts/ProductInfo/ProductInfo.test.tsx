import { render } from "@testing-library/react";

// Components
import ProductInfo from "./";

describe("Testing ProductInfo component", () => {
  it("Should match snapshot", () => {
    const component = render(<ProductInfo />);
    expect(component).toMatchSnapshot();
  });
});
