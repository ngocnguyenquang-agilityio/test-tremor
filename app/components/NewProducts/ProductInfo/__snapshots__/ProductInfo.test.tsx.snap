// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ProductInfo component Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="w-full p-4 bg-white dark:bg-dark-tremor-primary rounded-lg w-[67%] shadow-box-icon-default"
      >
        <p
          class="text-tremor-default text-primary dark:text-white font-bold text-xl mb-8"
        >
          Product Infomation
        </p>
        <div
          class="tremor-Flex-root flex w-full justify-between flex-col items-end"
        >
          <div
            class="tremor-Flex-root flex w-full justify-between items-start flex-col sm:flex-row"
          >
            <div
              class="tremor-Flex-root flex w-full justify-between items-center flex-col"
            >
              <div
                class="h-[70px] w-full min-w-[290px]"
              >
                <div
                  class="tremor-TextInput-root relative flex items-center min-w-[10rem] outline-none rounded-tremor-default transition duration-100 shadow-tremor-input dark:shadow-dark-tremor-input dark:hover:bg-dark-tremor-background-muted text-tremor-content border-tremor-brand-subtle ring-tremor-brand-muted dark:ring-dark-tremor-brand-muted border py-1 w-full dark:text-white bg-transparent dark:bg-transparent focus:bg-transparent rounded-b-none border-l-0 border-r-0 border-t-0 border-b-1 focus:border-b-2 focus:outline-none focus:border-tremor-brand-subtle dark:border-light dark:focus:border-white shadow-none hover:bg-transparent ring-0"
                >
                  <input
                    class="tremor-TextInput-input w-full focus:outline-none focus:ring-0 border-none bg-transparent text-tremor-default rounded-tremor-default transition duration-100 text-tremor-content-emphasis dark:text-dark-tremor-content-emphasis [appearance:textfield] [&::-webkit-outer-spin-button]:appearance-none [&::-webkit-inner-spin-button]:appearance-none pl-3 pr-4 py-2 placeholder:text-tremor-content dark:placeholder:text-dark-tremor-content"
                    data-testid="base-input"
                    id="name"
                    name="name"
                    placeholder="Name"
                    required=""
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class="tremor-Flex-root flex w-full justify-between items-center flex-col"
              >
                <div
                  class="tremor-Flex-root flex w-full flex-row items-center justify-start mb-3"
                >
                  <p
                    class="text-tremor-default text-secondary dark:text-white"
                  >
                    Description
                  </p>
                  <p
                    class="text-tremor-default text-tremor-content text-xs ml-1 dark:text-white"
                  >
                    (Optional)
                  </p>
                </div>
                <div
                  class="w-full min-w-[290px]"
                />
              </div>
            </div>
            <div
              class="tremor-Flex-root flex w-full justify-between items-center ml-0 flex-col mt-4 sm:mt-0 sm:ml-6"
            >
              <div
                class="h-[70px] w-full min-w-[290px]"
              >
                <div
                  class="tremor-TextInput-root relative flex items-center min-w-[10rem] outline-none rounded-tremor-default transition duration-100 shadow-tremor-input dark:shadow-dark-tremor-input dark:hover:bg-dark-tremor-background-muted text-tremor-content border-tremor-border border py-1 w-full dark:text-white bg-transparent dark:bg-transparent focus:bg-transparent rounded-b-none border-l-0 border-r-0 border-t-0 border-b-1 focus:border-b-2 focus:outline-none focus:border-tremor-brand-subtle dark:border-light dark:focus:border-white shadow-none hover:bg-transparent ring-0"
                >
                  <input
                    class="tremor-TextInput-input w-full focus:outline-none focus:ring-0 border-none bg-transparent text-tremor-default rounded-tremor-default transition duration-100 text-tremor-content-emphasis dark:text-dark-tremor-content-emphasis [appearance:textfield] [&::-webkit-outer-spin-button]:appearance-none [&::-webkit-inner-spin-button]:appearance-none pl-3 pr-4 py-2 placeholder:text-tremor-content dark:placeholder:text-dark-tremor-content"
                    data-testid="base-input"
                    id="weight"
                    name="weight"
                    placeholder="Weight"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class="tremor-Flex-root flex w-full justify-between flex-col items-start"
              >
                <p
                  class="text-tremor-default text-secondary mb-3 dark:text-white"
                >
                  Category
                </p>
                <div
                  class="h-[70px] w-full"
                >
                  <div
                    class="w-full min-w-[10rem] relative text-tremor-default select-custom dark:text-white dark:border-light dark:focus:border-white"
                    data-headlessui-state=""
                  >
                    <button
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      class="w-full outline-none text-left whitespace-nowrap truncate rounded-tremor-default focus:ring-2 transition duration-100 shadow-tremor-input focus:border-tremor-brand-subtle focus:ring-tremor-brand-muted dark:shadow-dark-tremor-input dark:focus:border-dark-tremor-brand-subtle dark:focus:ring-dark-tremor-brand-muted pl-3 pr-8 py-2 border bg-tremor-background dark:bg-dark-tremor-background hover:bg-tremor-background-muted dark:hover:bg-dark-tremor-background-muted text-tremor-content dark:text-dark-tremor-content border-tremor-border dark:border-dark-tremor-border"
                      data-headlessui-state=""
                      id="headlessui-listbox-button-:r0:"
                      type="button"
                    >
                      <span
                        class="w-[90%] block truncate"
                      >
                        Clothing
                      </span>
                      <span
                        class="absolute inset-y-0 right-0 flex items-center mr-3"
                      >
                        <svg
                          class="tremor-Select-arrowDownIcon flex-none text-tremor-content-subtle dark:text-dark-tremor-content-subtle h-4 w-4"
                          fill="none"
                          stroke="currentColor"
                          stroke-width="2.5"
                          viewBox="0 0 24 24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M19 9l-7 7-7-7"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                </div>
              </div>
              <div
                class="tremor-Flex-root flex w-full justify-between flex-col items-start"
              >
                <p
                  class="text-tremor-default text-secondary mb-3 dark:text-white"
                >
                  Size
                </p>
                <div
                  class="h-[70px] w-full"
                >
                  <div
                    class="w-full min-w-[10rem] relative text-tremor-default select-custom dark:text-white dark:bg-transparent"
                    data-headlessui-state=""
                  >
                    <button
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      class="w-full outline-none text-left whitespace-nowrap truncate rounded-tremor-default focus:ring-2 transition duration-100 shadow-tremor-input focus:border-tremor-brand-subtle focus:ring-tremor-brand-muted dark:shadow-dark-tremor-input dark:focus:border-dark-tremor-brand-subtle dark:focus:ring-dark-tremor-brand-muted pl-3 pr-8 py-2 border bg-tremor-background dark:bg-dark-tremor-background hover:bg-tremor-background-muted dark:hover:bg-dark-tremor-background-muted text-tremor-content dark:text-dark-tremor-content border-tremor-border dark:border-dark-tremor-border"
                      data-headlessui-state=""
                      id="headlessui-listbox-button-:r1:"
                      type="button"
                    >
                      <span
                        class="w-[90%] block truncate"
                      >
                        Small
                      </span>
                      <span
                        class="absolute inset-y-0 right-0 flex items-center mr-3"
                      >
                        <svg
                          class="tremor-Select-arrowDownIcon flex-none text-tremor-content-subtle dark:text-dark-tremor-content-subtle h-4 w-4"
                          fill="none"
                          stroke="currentColor"
                          stroke-width="2.5"
                          viewBox="0 0 24 24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M19 9l-7 7-7-7"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <button
            class="tremor-Button-root flex-shrink-0 inline-flex justify-center items-center group font-medium outline-none rounded-tremor-default shadow-tremor-input dark:shadow-dark-tremor-input text-xs border-tremor-brand dark:border-dark-tremor-brand text-tremor-brand-inverted dark:text-dark-tremor-brand-inverted hover:bg-tremor-brand-emphasis dark:hover:bg-dark-tremor-brand-emphasis hover:border-tremor-brand-emphasis dark:hover:border-dark-tremor-brand-emphasis antialiased min-w-[80px] text-center uppercase sm:px-[21px] bg-gradient-primary dark:bg-gradient-pickled !shadow-btn-primary px-6 py-2.5 rounded-lg border-0 hover:!shadow-btn-primary-hover"
            type="submit"
          >
            <span
              class="tremor-Button-text text-sm whitespace-nowrap"
            >
              <p
                class="text-tremor-default items-center uppercase py-[2px] text-xs font-bold font-primary text-white dark:text-dark-tremor-content-title"
              >
                Next
              </p>
            </span>
          </button>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="w-full p-4 bg-white dark:bg-dark-tremor-primary rounded-lg w-[67%] shadow-box-icon-default"
    >
      <p
        class="text-tremor-default text-primary dark:text-white font-bold text-xl mb-8"
      >
        Product Infomation
      </p>
      <div
        class="tremor-Flex-root flex w-full justify-between flex-col items-end"
      >
        <div
          class="tremor-Flex-root flex w-full justify-between items-start flex-col sm:flex-row"
        >
          <div
            class="tremor-Flex-root flex w-full justify-between items-center flex-col"
          >
            <div
              class="h-[70px] w-full min-w-[290px]"
            >
              <div
                class="tremor-TextInput-root relative flex items-center min-w-[10rem] outline-none rounded-tremor-default transition duration-100 shadow-tremor-input dark:shadow-dark-tremor-input dark:hover:bg-dark-tremor-background-muted text-tremor-content border-tremor-brand-subtle ring-tremor-brand-muted dark:ring-dark-tremor-brand-muted border py-1 w-full dark:text-white bg-transparent dark:bg-transparent focus:bg-transparent rounded-b-none border-l-0 border-r-0 border-t-0 border-b-1 focus:border-b-2 focus:outline-none focus:border-tremor-brand-subtle dark:border-light dark:focus:border-white shadow-none hover:bg-transparent ring-0"
              >
                <input
                  class="tremor-TextInput-input w-full focus:outline-none focus:ring-0 border-none bg-transparent text-tremor-default rounded-tremor-default transition duration-100 text-tremor-content-emphasis dark:text-dark-tremor-content-emphasis [appearance:textfield] [&::-webkit-outer-spin-button]:appearance-none [&::-webkit-inner-spin-button]:appearance-none pl-3 pr-4 py-2 placeholder:text-tremor-content dark:placeholder:text-dark-tremor-content"
                  data-testid="base-input"
                  id="name"
                  name="name"
                  placeholder="Name"
                  required=""
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="tremor-Flex-root flex w-full justify-between items-center flex-col"
            >
              <div
                class="tremor-Flex-root flex w-full flex-row items-center justify-start mb-3"
              >
                <p
                  class="text-tremor-default text-secondary dark:text-white"
                >
                  Description
                </p>
                <p
                  class="text-tremor-default text-tremor-content text-xs ml-1 dark:text-white"
                >
                  (Optional)
                </p>
              </div>
              <div
                class="w-full min-w-[290px]"
              />
            </div>
          </div>
          <div
            class="tremor-Flex-root flex w-full justify-between items-center ml-0 flex-col mt-4 sm:mt-0 sm:ml-6"
          >
            <div
              class="h-[70px] w-full min-w-[290px]"
            >
              <div
                class="tremor-TextInput-root relative flex items-center min-w-[10rem] outline-none rounded-tremor-default transition duration-100 shadow-tremor-input dark:shadow-dark-tremor-input dark:hover:bg-dark-tremor-background-muted text-tremor-content border-tremor-border border py-1 w-full dark:text-white bg-transparent dark:bg-transparent focus:bg-transparent rounded-b-none border-l-0 border-r-0 border-t-0 border-b-1 focus:border-b-2 focus:outline-none focus:border-tremor-brand-subtle dark:border-light dark:focus:border-white shadow-none hover:bg-transparent ring-0"
              >
                <input
                  class="tremor-TextInput-input w-full focus:outline-none focus:ring-0 border-none bg-transparent text-tremor-default rounded-tremor-default transition duration-100 text-tremor-content-emphasis dark:text-dark-tremor-content-emphasis [appearance:textfield] [&::-webkit-outer-spin-button]:appearance-none [&::-webkit-inner-spin-button]:appearance-none pl-3 pr-4 py-2 placeholder:text-tremor-content dark:placeholder:text-dark-tremor-content"
                  data-testid="base-input"
                  id="weight"
                  name="weight"
                  placeholder="Weight"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="tremor-Flex-root flex w-full justify-between flex-col items-start"
            >
              <p
                class="text-tremor-default text-secondary mb-3 dark:text-white"
              >
                Category
              </p>
              <div
                class="h-[70px] w-full"
              >
                <div
                  class="w-full min-w-[10rem] relative text-tremor-default select-custom dark:text-white dark:border-light dark:focus:border-white"
                  data-headlessui-state=""
                >
                  <button
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    class="w-full outline-none text-left whitespace-nowrap truncate rounded-tremor-default focus:ring-2 transition duration-100 shadow-tremor-input focus:border-tremor-brand-subtle focus:ring-tremor-brand-muted dark:shadow-dark-tremor-input dark:focus:border-dark-tremor-brand-subtle dark:focus:ring-dark-tremor-brand-muted pl-3 pr-8 py-2 border bg-tremor-background dark:bg-dark-tremor-background hover:bg-tremor-background-muted dark:hover:bg-dark-tremor-background-muted text-tremor-content dark:text-dark-tremor-content border-tremor-border dark:border-dark-tremor-border"
                    data-headlessui-state=""
                    id="headlessui-listbox-button-:r0:"
                    type="button"
                  >
                    <span
                      class="w-[90%] block truncate"
                    >
                      Clothing
                    </span>
                    <span
                      class="absolute inset-y-0 right-0 flex items-center mr-3"
                    >
                      <svg
                        class="tremor-Select-arrowDownIcon flex-none text-tremor-content-subtle dark:text-dark-tremor-content-subtle h-4 w-4"
                        fill="none"
                        stroke="currentColor"
                        stroke-width="2.5"
                        viewBox="0 0 24 24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M19 9l-7 7-7-7"
                          stroke-linecap="round"
                          stroke-linejoin="round"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
              </div>
            </div>
            <div
              class="tremor-Flex-root flex w-full justify-between flex-col items-start"
            >
              <p
                class="text-tremor-default text-secondary mb-3 dark:text-white"
              >
                Size
              </p>
              <div
                class="h-[70px] w-full"
              >
                <div
                  class="w-full min-w-[10rem] relative text-tremor-default select-custom dark:text-white dark:bg-transparent"
                  data-headlessui-state=""
                >
                  <button
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    class="w-full outline-none text-left whitespace-nowrap truncate rounded-tremor-default focus:ring-2 transition duration-100 shadow-tremor-input focus:border-tremor-brand-subtle focus:ring-tremor-brand-muted dark:shadow-dark-tremor-input dark:focus:border-dark-tremor-brand-subtle dark:focus:ring-dark-tremor-brand-muted pl-3 pr-8 py-2 border bg-tremor-background dark:bg-dark-tremor-background hover:bg-tremor-background-muted dark:hover:bg-dark-tremor-background-muted text-tremor-content dark:text-dark-tremor-content border-tremor-border dark:border-dark-tremor-border"
                    data-headlessui-state=""
                    id="headlessui-listbox-button-:r1:"
                    type="button"
                  >
                    <span
                      class="w-[90%] block truncate"
                    >
                      Small
                    </span>
                    <span
                      class="absolute inset-y-0 right-0 flex items-center mr-3"
                    >
                      <svg
                        class="tremor-Select-arrowDownIcon flex-none text-tremor-content-subtle dark:text-dark-tremor-content-subtle h-4 w-4"
                        fill="none"
                        stroke="currentColor"
                        stroke-width="2.5"
                        viewBox="0 0 24 24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M19 9l-7 7-7-7"
                          stroke-linecap="round"
                          stroke-linejoin="round"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <button
          class="tremor-Button-root flex-shrink-0 inline-flex justify-center items-center group font-medium outline-none rounded-tremor-default shadow-tremor-input dark:shadow-dark-tremor-input text-xs border-tremor-brand dark:border-dark-tremor-brand text-tremor-brand-inverted dark:text-dark-tremor-brand-inverted hover:bg-tremor-brand-emphasis dark:hover:bg-dark-tremor-brand-emphasis hover:border-tremor-brand-emphasis dark:hover:border-dark-tremor-brand-emphasis antialiased min-w-[80px] text-center uppercase sm:px-[21px] bg-gradient-primary dark:bg-gradient-pickled !shadow-btn-primary px-6 py-2.5 rounded-lg border-0 hover:!shadow-btn-primary-hover"
          type="submit"
        >
          <span
            class="tremor-Button-text text-sm whitespace-nowrap"
          >
            <p
              class="text-tremor-default items-center uppercase py-[2px] text-xs font-bold font-primary text-white dark:text-dark-tremor-content-title"
            >
              Next
            </p>
          </span>
        </button>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
